app/controllers/api/submission_files_controller.rb

```diff
@@ -199,10 +163,91 @@ def remove_folder
       end
     end
 
+    def submit_file
+      student = Student.find_by(user: @real_user, course: @current_course)
+      assignment = Assignment.find_by(id: params[:assignment_id], course: @current_course)
+
+      # Disable submission via API if the instructor desires to
+      unless assignment.api_submit
+        render 'shared/http_status', locals: { code: '403', message:
+          'The instructor has disabled submission via the API' }, status: :forbidden
+        return
+      end
+
+      # Reject submission if the student has pending grouping
```

Comment:
This isn't a bad idea, but in this case I think it's better to just allow students to create their own group (deleting pending group invitations, which I think will happen automatically from callbacks). Remember that students may be submitting very close to the deadline, and the point of the Jupyter extension is to get rid of as much friction as possible.